src/main/java/com/github/leroyguillaume/keycloak/bcrypt/BCryptPasswordHashProvider.java

```diff
@@ -37,7 +39,7 @@ public PasswordCredentialModel encodedCredential(final String rawPassword, final
     @Override
     public String encode(final String rawPassword, final int iterations) {
         final int cost = iterations == -1 ? defaultIterations : iterations;
-        return BCrypt.with(BCrypt.Version.VERSION_2A).hashToString(cost, rawPassword.toCharArray());
+        return BCrypt.with(LongPasswordStrategies.hashSha512(BCrypt.Version.VERSION_2Y_NO_NULL_TERMINATOR)).hashToString(cost, rawPassword.toCharArray());
```

Comment:
I think we shouldn't pre-hash, it's not recommended according to [owasp](https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#pre-hashing-passwords-with-bcrypt). Let's just truncate which is what's how it's done today in the gem.